docs/aws/audit/appsyncmonitoring/rules/appsync_authorization.mdx

```diff
@@ -24,6 +24,163 @@ CBP
+   If you have an existing AppSync API without authorization, you can update it to include the desired authorization type.
+
+   ```sh
+   aws appsync update-graphql-api \
+     --api-id <your-api-id> \
+     --name "MyUpdatedAppSyncAPI" \
+     --authentication-type "AWS_IAM"
+   ```
+
+3. **Configure API Key Expiration:**
+   If you are using API keys for authorization, ensure that the API keys have an expiration date to enhance security.
+
+   ```sh
+   aws appsync create-api-key \
+     --api-id <your-api-id> \
+     --description "My API Key" \
+     --expires <expiration-timestamp>
+   ```
+
+4. **Add Additional Authorization Providers:**
+   You can add multiple authorization providers to your AppSync API for more robust security. For example, adding Amazon Cognito User Pools.
+
+   ```sh
+   aws appsync update-graphql-api \
+     --api-id <your-api-id> \
+     --additional-authentication-providers "authenticationType=AMAZON_COGNITO_USER_POOLS,userPoolConfig={userPoolId=<your-user-pool-id>,awsRegion=<your-region>,appIdClientRegex=<your-app-client-id>}"
+   ```
+
+By following these steps, you can ensure that your AppSync APIs have proper authorization configurations, thereby preventing unauthorized access.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent AppSync APIs from lacking authorization configuration in AWS AppSync using Python scripts, you can use the AWS SDK for Python (Boto3). Here are the steps to ensure that your AppSync APIs have proper authorization configurations:
+
+1. **Install Boto3**:
+   Ensure you have Boto3 installed in your Python environment. If not, you can install it using pip:
+   ```bash
+   pip install boto3
+   ```
+
+2. **Set Up AWS Credentials**:
+   Make sure your AWS credentials are configured. You can set them up using the AWS CLI or by setting environment variables:
+   ```bash
+   aws configure
+   ```
+   Alternatively, you can set the environment variables directly in your script:
+   ```python
+   import os
+   os.environ['AWS_ACCESS_KEY_ID'] = 'your_access_key_id'
+   os.environ['AWS_SECRET_ACCESS_KEY'] = 'your_secret_access_key'
+   os.environ['AWS_DEFAULT_REGION'] = 'your_region'
+   ```
+
+3. **Create or Update AppSync API with Authorization Configuration**:
+   Use Boto3 to create or update an AppSync API with the necessary authorization configuration. Here’s a sample script to ensure that an AppSync API has authorization configured:
+
+   ```python
+   import boto3
+
+   # Initialize the AppSync client
+   client = boto3.client('appsync')
+
+   # Define the authorization configuration
+   auth_config = {
+       'authorizationType': 'API_KEY',  # or 'AWS_IAM', 'OPENID_CONNECT', 'AMAZON_COGNITO_USER_POOLS'
+       'apiKeyConfig': {
+           'expires': '2023-12-31T23:59:59Z'  # Example expiration date for API key
+       }
+   }
+
+   # Function to create or update AppSync API with authorization
+   def ensure_appsync_auth(api_id, auth_config):
+       try:
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>